libraries/hotstuff/chain_pacemaker.cpp

```diff
@@ -216,19 +218,62 @@ namespace eosio { namespace hotstuff {
 
    // called from main thread
    void chain_pacemaker::on_accepted_block( const block_state_ptr& blk ) {
-      std::scoped_lock g( _chain_state_mutex );
-      _head_block_state = blk;
+      {
+         std::scoped_lock g( _chain_state_mutex );
+         _head_block_state = blk;
+      }
+      if (!blk->block->header_extensions.empty()) {
+         std::optional<block_header_extension> ext = blk->block->extract_header_extension(hs_finalizer_set_extension::extension_id());
+         if (ext) {
+            _commitment_mgr.store_finset_proposal(blk, std::move(std::get<hs_finalizer_set_extension>(*ext)));
+         }
+      }
+      signed_block_ptr sb = blk->block;
+      std::optional<block_extension> ext = sb->extract_extension(hs_commitment_extension::extension_id());
```

Comment:
I'm not sure that it is not needed, since we need to track the finalizer_set with a delay (when seen in block), which is different from the active finalizer_set of hotstuff. Also, the optimization is not significant because most blocks will not have commitments.